genrl/core/rollout_storage.py

```diff
@@ -310,3 +257,66 @@ def _get_samples(self, batch_inds: np.ndarray) -> RolloutBufferSamples:
             self.returns[batch_inds].flatten(),
         )
         return RolloutBufferSamples(*tuple(map(self.to_torch, data)))
+
+
+def compute_returns_and_advantage(
+    rollout_buffer: Union[RolloutBuffer, BaseBuffer],
+    last_value: torch.Tensor,
+    dones: np.ndarray,
+    use_gae: bool = False,
```

Comment:
Put a little more thought into this. Its trivial to separate the function out directly.